src/sb_pointcloud_processing/src/rgb_to_hsv.cpp

```diff
@@ -0,0 +1,36 @@
+/**
+ * created by: valerian ratu
+ * Created on: 2018/01/13
+ * Description: A ros nodelet which takes pointcloud input in RGB colourspace
+ *              and returns it in the HSV colourspace.
+ */
+
+#include <pluginlib/class_list_macros.h>
+#include <rgb_to_hsv.h>
+
+using namespace sb_pointcloud_processing;
+
+RGBtoHSV::RGBtoHSV() {}
+
+void RGBtoHSV::onInit() {
+    NODELET_DEBUG("Initializing Nodelet...");
+    ros::NodeHandle& private_nh = getPrivateNodeHandle();
+    sub = private_nh.subscribe("input", 1, &RGBtoHSV::callback, this);
+    pub = private_nh.advertise<pcl::PointCloud<pcl::PointXYZHSV>>("output", 1);
+    NODELET_DEBUG("Nodelet Initialized");
+}
+
+void RGBtoHSV::callback(const sensor_msgs::PointCloud2::ConstPtr& input) {
+    pcl::PCLPointCloud2::Ptr pcl_input(new pcl::PCLPointCloud2);
```

Comment:
pcl to pcl2 and pcl2 to pcl conversions should be a function.

Reply:
same as above